node/src/main/scala/com/wavesplatform/api/http/utils/UtilsEvaluator.scala

```diff
@@ -0,0 +1,103 @@
+import com.wavesplatform.lang.directives.values.{DApp as DAppType, *}
+import com.wavesplatform.lang.script.Script
+import com.wavesplatform.lang.v1.compiler.Terms.{EVALUATED, EXPR}
+import com.wavesplatform.lang.v1.compiler.{ContractScriptCompactor, ExpressionCompiler, Terms}
+import com.wavesplatform.lang.v1.evaluator.ContractEvaluator.Invocation
+import com.wavesplatform.lang.v1.evaluator.{ContractEvaluator, EvaluatorV2, Log}
+import com.wavesplatform.lang.v1.traits.Environment.Tthis
+import com.wavesplatform.lang.v1.traits.domain.Recipient
+import com.wavesplatform.lang.v1.{ContractLimits, FunctionHeader}
+import com.wavesplatform.lang.{ValidationError, utils}
+import com.wavesplatform.state.diffs.invoke.InvokeScriptTransactionLike
+import com.wavesplatform.state.{Blockchain, Diff}
+import com.wavesplatform.transaction.Asset.IssuedAsset
+import com.wavesplatform.transaction.TransactionType.TransactionType
+import com.wavesplatform.transaction.TxValidationError.{GenericError, ScriptExecutionError}
+import com.wavesplatform.transaction.smart.*
+import com.wavesplatform.transaction.smart.InvokeScriptTransaction.Payment
+import com.wavesplatform.transaction.{Asset, TransactionType}
+import monix.eval.Coeval
+import shapeless.Coproduct
+
+object UtilsEvaluator {
+  def compile(version: StdLibVersion)(str: String): Either[GenericError, EXPR] =
+    ExpressionCompiler
+      .compileUntyped(str, utils.compilerContext(version, Expression, isAssetScript = false).copy(arbitraryDeclarations = true))
+      .leftMap(GenericError(_))
+
+  def toExpr(script: Script, invocation: Invocation): Either[ValidationError, EXPR] =
+    ContractEvaluator
+      .buildExprFromInvocation(script.expr.asInstanceOf[DApp], invocation, script.stdLibVersion)
+      .leftMap(e => GenericError(e.message))
+
+  def executeExpression(blockchain: Blockchain, script: Script, address: Address, pk: PublicKey, limit: Int)(
+      expr: EXPR
+  ): Either[ValidationError, (EVALUATED, Int, Log[Id])] =
+    for {
+      ds <- DirectiveSet(script.stdLibVersion, Account, DAppType).leftMap(GenericError(_))
+      invoke = new InvokeScriptTransactionLike {
+        override def dApp: AddressOrAlias              = address
+        override def funcCall: Terms.FUNCTION_CALL     = Terms.FUNCTION_CALL(FunctionHeader.User(""), Nil)
+        override def payments: Seq[Payment]            = Seq.empty
+        override def root: InvokeScriptTransactionLike = this
+        override val sender: PublicKey                 = PublicKey(ByteStr(new Array[Byte](32)))
+        override def assetFee: (Asset, Long)           = Asset.Waves -> 0L
+        override def timestamp: Long                   = System.currentTimeMillis()
+        override def chainId: Byte                     = AddressScheme.current.chainId
+        override def id: Coeval[ByteStr]               = Coeval.evalOnce(ByteStr.empty)
+        override def checkedAssets: Seq[IssuedAsset]   = Seq.empty
```

Comment:
This actually is a sequence of scripted assets. So, their scripts won't run.
Am I right?

Reply:
Will be fixed in another task